src/geometry.F90

```diff
@@ -1907,5 +1885,82 @@ recursive subroutine count_instance(univ)
 
   end subroutine count_instance
 
+!===============================================================================
+! MAXIMUM_LEVELS determines the maximum number of nested coordinate levels in
+! the geometry
+!===============================================================================
+
+  recursive function maximum_levels(univ) result(levels)
+
+    type(Universe), intent(in) :: univ  ! universe to search through
+    integer :: levels                   ! maximum number of levels for this universe
+
+    integer :: i                          ! index over cells
+    integer :: j, k, m                    ! indices in lattice
+    integer :: levels_below               ! max levels below this universe
+    type(Cell),     pointer :: c          ! pointer to current cell
+    type(Universe), pointer :: next_univ  ! next universe to loop through
+    class(Lattice), pointer :: lat        ! pointer to current lattice
+
+    levels_below = 0
+    do i = 1, univ % n_cells
+      c => cells(univ % cells(i))
+
+      ! ====================================================================
+      ! CELL CONTAINS LOWER UNIVERSE, RECURSIVELY FIND CELL
+      if (c % type == CELL_FILL) then
+
+        next_univ => universes(c % fill)
+        levels_below = maximum_levels(next_univ)
```

Comment:
Should this not be:

```fortran
levels_below = max(levels_below, maximum_levels(next_univ))
```

as it is below for ``CELL_LATTICE`` types? Otherwise I think this will fail if it finds a ``CELL_FILL`` with fewer levels *after* inspecting a ``CELL_LATTICE`` with more levels.